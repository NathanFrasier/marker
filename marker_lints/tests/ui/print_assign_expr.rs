// normalize-stdout-windows: "tests/ui/" -> "$$DIR/"

fn bar() -> i32 {
    16
}

pub fn main() {
    let mut a = 0;
    let mut b = 0;
    let _print_exprs = {
        a = bar();
        a += 1;
        [a, b] = [1, 2];
        ()
    };
}
