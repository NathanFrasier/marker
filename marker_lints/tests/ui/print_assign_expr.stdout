Block(
    BlockExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        stmts: [
            Expr(
                Assign(
                    AssignExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        assignee: Path(
                            PathExpr {
                                data: CommonExprData {
                                    _lifetime: PhantomData<&()>,
                                    id: ExprId(..),
                                    span: SpanId(..),
                                },
                                path: AstQPath {
                                    self_ty: None,
                                    path_ty: None,
                                    path: AstPath {
                                        segments: [
                                            AstPathSegment {
                                                ident: Ident {
                                                    name: "x",
                                                    span: Span {
                                                        source: File(
                                                            "$DIR/print_assign_expr.rs",
                                                        ),
                                                        start: 150,
                                                        end: 151,
                                                    },
                                                },
                                                generics: GenericArgs {
                                                    args: [],
                                                },
                                            },
                                        ],
                                    },
                                    target: Var(
                                        VarId(..),
                                    ),
                                },
                            },
                        ),
                        value: Call(
                            CallExpr {
                                data: CommonExprData {
                                    _lifetime: PhantomData<&()>,
                                    id: ExprId(..),
                                    span: SpanId(..),
                                },
                                operand: Path(
                                    PathExpr {
                                        data: CommonExprData {
                                            _lifetime: PhantomData<&()>,
                                            id: ExprId(..),
                                            span: SpanId(..),
                                        },
                                        path: AstQPath {
                                            self_ty: None,
                                            path_ty: None,
                                            path: AstPath {
                                                segments: [
                                                    AstPathSegment {
                                                        ident: Ident {
                                                            name: "bar",
                                                            span: Span {
                                                                source: File(
                                                                    "$DIR/print_assign_expr.rs",
                                                                ),
                                                                start: 154,
                                                                end: 157,
                                                            },
                                                        },
                                                        generics: GenericArgs {
                                                            args: [],
                                                        },
                                                    },
                                                ],
                                            },
                                            target: Item(
                                                ItemId(..),
                                            ),
                                        },
                                    },
                                ),
                                args: [],
                            },
                        ),
                        op: None,
                    },
                ),
            ),
            Expr(
                Assign(
                    AssignExpr {
                        data: CommonExprData {
                            _lifetime: PhantomData<&()>,
                            id: ExprId(..),
                            span: SpanId(..),
                        },
                        assignee: Path(
                            PathExpr {
                                data: CommonExprData {
                                    _lifetime: PhantomData<&()>,
                                    id: ExprId(..),
                                    span: SpanId(..),
                                },
                                path: AstQPath {
                                    self_ty: None,
                                    path_ty: None,
                                    path: AstPath {
                                        segments: [
                                            AstPathSegment {
                                                ident: Ident {
                                                    name: "x",
                                                    span: Span {
                                                        source: File(
                                                            "$DIR/print_assign_expr.rs",
                                                        ),
                                                        start: 169,
                                                        end: 170,
                                                    },
                                                },
                                                generics: GenericArgs {
                                                    args: [],
                                                },
                                            },
                                        ],
                                    },
                                    target: Var(
                                        VarId(..),
                                    ),
                                },
                            },
                        ),
                        value: IntLit(
                            IntLitExpr {
                                data: CommonExprData {
                                    _lifetime: PhantomData<&()>,
                                    id: ExprId(..),
                                    span: SpanId(..),
                                },
                                value: 1,
                                suffix: None,
                            },
                        ),
                        op: Some(
                            Add,
                        ),
                    },
                ),
            ),
        ],
        expr: Some(
            Tuple(
                TupleExpr {
                    data: CommonExprData {
                        _lifetime: PhantomData<&()>,
                        id: ExprId(..),
                        span: SpanId(..),
                    },
                    elements: [],
                },
            ),
        ),
        is_unsafe: false,
    },
)

